app/models/reservation.rb

```diff
@@ -103,6 +103,25 @@ class Reservation < ActiveRecord::Base
       .select('reservations.*, equipment_models.category_id as category_id')
   }
 
+  # for status modifying jobs
+  def self.deletable_missed
+    return Reservation.none if AppConfig.check(:res_exp_time, '').blank?
+    threshold = Time.zone.today - AppConfig.get(:res_exp_time).days
+    Reservation.missed.where('start_date < ?', threshold)
+  end
+
+  def self.missed_not_emailed
+    Reservation.missed.not_flagged(:missed_email_sent)
+  end
+
+  def self.newly_missed
```

Comment:
Same as below; wouldn't this return all reservations whose start dates had passed, even those that had already had their status changed to `missed`?